breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BreadthFirstSearch <T> {
+
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<Integer, ArrayList<T>> adj;
```

Comment:
This will not start working by just modifying a single line. You'll have to rethink the situation here. You can't call methods like `initEdges` now because they aren't generic. This method was always a bad idea.

You'll have to tackle the situation that user may be adding an edge between two unregistered nodes.